plumbing/format/packfile/delta_selector.go

```diff
@@ -231,7 +231,8 @@ func (dw *deltaSelector) walk(
 			delete(indexMap, obj.Hash())
 
 			if obj.IsDelta() {
-				obj.Original = nil
+				obj.SetOriginal(obj.Original)
+				obj.CleanOriginal()
```

Comment:
this is a little bit strange, maybe is worth it to add a method that just add the original metadata to the objectToPack, something like `obj.SetOriginalMetadata(obj.Original)`

Reply:
I've adede a `SaveOriginalMetadata` function. It uses current `Original` object.